whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java

```diff
@@ -46,6 +54,26 @@ public AbuseCFinder(@Qualifier("jdbcRpslObjectSlaveDao") final RpslObjectDao obj
         this.ipv6Tree = ipv6Tree;
         this.maintainers = maintainers;
         this.abuseValidationStatusDao = abuseValidationStatusDao;
+        this.ripeSources = getRipeSources(mainSource, nonAuthSource, grsSource);
+    }
+
+    private ImmutableSet<CIString> getRipeSources(final String mainSource, final String nonAuthSource, final String grsSource) {
+        ImmutableSet.Builder<CIString> sourceBuilder  = new ImmutableSet.Builder<>();
+
+        if(StringUtils.isNotEmpty(mainSource)) {
```

Comment:
whois.source is mandatory (although it could be empty it's unlikely!), no need for a check.